### PR TITLE
Update documentation with the OAuth App authentication capability 

### PR DESCRIPTION
**Pull request by dimitarproynov**
Testing done:
NA

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

### CODE REVIEW

File: website/docs/index.html.markdown [hunk 1]
```diff
@@ -21,18 +21,22 @@ resources.
...
+Note that in all the examples you will need to update the `api_token` (or `client_id` and `client_secret`)
+and `org_id` settings in the variables.tf file to match those configured in your VMC environment.
 
 
 ## Argument Reference
 
 The following arguments are used to configure the VMware Cloud on AWS Provider:
 
-* `api_token` - (Required) API token is used to authenticate when calling VMware Cloud Services APIs.
+* `api_token` - (Required, in conflict with "client_id" and "client_secret") API token is used to authenticate when calling VMware Cloud Services APIs.
```
Comment (karaatanassov) [inline website/docs/index.html.markdown:34]: "in conflict with" is a bit on the offensive side. You can say "alternative to"

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -21,18 +21,22 @@ resources.
 In order to use the provider you need to obtain the authentication
 token from the Cloud Service Provider by providing the org scoped API token.
 The provider client uses Cloud Service Provider (CSP) API
-to exchange this org scoped API token for user access token.
+to exchange this org scoped API token/OAuth App Id and Secret for user access token.
 
-Note that in all of the examples you will need to update the `api_token` and `org_id` settings
-in the variables.tf file to match those configured in your VMC environment.
+Note that in all the examples you will need to update the `api_token` (or `client_id` and `client_secret`)
```
Comment (karaatanassov) [inline website/docs/index.html.markdown]: I do not think you need the brackets. I guess a lot of readers will be developers and can benefit the brackets for logical grouping. It is not easy to read them though.

File: website/docs/index.html.markdown [hunk 3]
```diff
@@ -21,18 +21,22 @@ resources.
...
 ## Argument Reference
 
 The following arguments are used to configure the VMware Cloud on AWS Provider:
 
-* `api_token` - (Required) API token is used to authenticate when calling VMware Cloud Services APIs.
+* `api_token` - (Required, in conflict with "client_id" and "client_secret") API token is used to authenticate when calling VMware Cloud Services APIs.
    This token is scoped within the organization.
+* `client_id` - (Required in pair with "client_secret", in conflict with "api_token") ID of OAuth App associated with the organization. The combination with
+   "client_secret" is used to authenticate when calling VMware Cloud Services APIs.
+* `client_secret` - (Required in pair with "client_id", in conflict with "api_token") Secret of OAuth App associated with the organization. The combination with
```
Comment (karaatanassov) [inline website/docs/index.html.markdown:38]: "in conflict with" -> "alternative to"

**Review by karaatanassov — APPROVED** [review]
Looks good except the "in conflict with". If you can use a bit softer language.

File: website/docs/index.html.markdown (hunk 1 above)
Comment (dimitarproynov) [reply website/docs/index.html.markdown:34]: This is the term the terraform SDK uses. See:
https://github.com/vmware/terraform-provider-vmc/blob/main/vmc/provider.go#L21

File: website/docs/index.html.markdown (hunk 3 above)
Comment (dimitarproynov) [reply website/docs/index.html.markdown:38]: This is the term the terraform SDK uses. See:
https://github.com/vmware/terraform-provider-vmc/blob/main/vmc/provider.go#L21

File: website/docs/index.html.markdown (hunk 2 above)
Comment (dimitarproynov) [reply website/docs/index.html.markdown]: Addressed.

**Pull request merged**